napari/utils/geometry.py

```diff
@@ -332,15 +332,53 @@ def intersect_line_with_plane_3d(
 
     # project direction between line and plane onto the plane normal
     line_plane_direction = plane_position - line_position
-    line_plane_on_plane_normal = np.dot(line_plane_direction, plane_normal)
+    line_plane_on_plane_normal = np.sum(
+        line_plane_direction * plane_normal, axis=1
+    )
 
     # project line direction onto the plane normal
-    line_direction_on_plane_normal = np.dot(line_direction, plane_normal)
+    line_direction_on_plane_normal = np.sum(
+        line_direction * plane_normal, axis=1
+    )
 
     # find scale factor for line direction
     scale_factor = line_plane_on_plane_normal / line_direction_on_plane_normal
 
-    return line_position + (scale_factor * line_direction)
+    if plane_position.ndim == 2:
```

Comment:
I am not sure we will be able to nicely handle doing one plane and an array of planes in the same function. I had to do this funky stuff because the `scale_factor` is a scalar value for the one plane case and is an array of values for the many planes case. Perhaps with some more clever numpy we can get everything to broadcast nicely. Otherwise, we can just make two functions.

Reply:
`np.atleast_2d`?